### PR TITLE
Feature/nijo 364 locations normalization

### PR DESCRIPTION
**Pull request by hasmikharutyunyan**

### CODE REVIEW

File: LocationsNamesNormalizer/Extensions/StringExtension.cs [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+        {
+            var textInfo = new CultureInfo(DefaultCultureName).TextInfo;
+
+            return textInfo.ToTitleCase(value.ToLower());
+        }
+
+
+        public static string ToHtmlDecoded(this string value) => WebUtility.HtmlDecode(value);
+
+        public static string ToStringWithoutSpecialCharacters(this string value) => Regex.Replace(value, SpecialCharactersProcessingPattern, "");
```
Comment (kirillta) [inline LocationsNamesNormalizer/Extensions/StringExtension.cs]: You may compile the regex

File: LocationsNamesNormalizer/LocationsNamesNormalizer.csproj [hunk 2]
```diff
@@ -0,0 +1,22 @@
+<Project Sdk="Microsoft.NET.Sdk">
+
+  <PropertyGroup>
+    <TargetFramework>netcoreapp3.1</TargetFramework>
+    <Version>1.0</Version>
+    <PackageReleaseNotes>Adeed names normalizator and default locations selector.</PackageReleaseNotes>
```
Comment (mingazhev) [inline LocationsNamesNormalizer/LocationsNamesNormalizer.csproj]: typo

File: LocationsNamesNormalizer/LocationsNamesNormalizer.csproj [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+    <PackageVersion>1.0</PackageVersion>
+  </PropertyGroup>
+
+  <ItemGroup>
+    <None Remove="Countries.json" />
+    <EmbeddedResource Include="Countries.json">
+      <CopyToOutputDirectory>Always</CopyToOutputDirectory>
```
Comment (mingazhev) [inline LocationsNamesNormalizer/LocationsNamesNormalizer.csproj]: is this line needed?

File: LocationsNamesNormalizer/LocationNamesRetriever.cs [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+
+namespace LocationsNamesNormalizer
+{
+    public class LocationNamesRetriever : ILocationNamesRetriever
+    {
+        public List<Country> RetrieveCountries()
+        {
+            try
+            {
+                var json = File.ReadAllText(CountriesFilePath);
```
Comment (mingazhev) [inline LocationsNamesNormalizer/LocationNamesRetriever.cs]: Still getting values from file. Need to change this to get data from Embedded resource. There are some examples there:

https://stackoverflow.com/questions/3314140/how-to-read-embedded-resource-text-file

File: LocationsNamesNormalizer/LocationNamesRetriever.cs [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+                return serializer.Deserialize<List<Country>>(jsonTextReader);
+            }
+            catch
+            {
+                return new List<Country>();
+            }
+        }
+
+
+        private const string CountriesFilePath = "LocationsNamesNormalizer.Countries.json";
```
Comment (mingazhev) [inline LocationsNamesNormalizer/LocationNamesRetriever.cs]: [Optional] this is not path, it is a resource key

**Pull request merged**